### PR TITLE
feat(3234): Improve metadata functional tests

### PR DESCRIPTION
**Pull request by VonnyJap**
## Context

The set of functional tests that are available right now to test metadata behavior is very limited. 

## Objective

This PR adds more functional tests to ensure the metadata behavior is working as expected. 

## References

https://github.com/screwdriver-cd/screwdriver/issues/3234

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: features/metadata.feature [hunk 1]
```diff
@@ -38,22 +38,42 @@ Feature: Metadata
...
+            | job   | triggers  |
+            | main  | BAR       |
+            | BAR   | BAZ       |
+            | BAZ   | BOOZ      |
+            | BAM   |           |
+        And the "BOOZ" job is disabled
         When the "main" job is started
-        Then add the { "foo": <foobar> } to metadata in the "main" build container
         And the build succeeded
+        Then add the { "foo": <foobar> } to metadata in the "main" build container
```
Comment (tkyi) [inline features/metadata.feature:49]: should this be moved back up to before `And the build succeeded`?

File: features/metadata.feature [hunk 2]
```diff
@@ -38,22 +38,42 @@ Feature: Metadata
...
         And the "BAR" job is started
         Then in the build, the { "foo": <foobar> } is available from metadata
-        And add the { "bar": <barbaz> } to metadata in the "BAR" build container
         And the build succeeded
+        And add the { "bar": <barbaz> } to metadata in the "BAR" build container
```
Comment (tkyi) [inline features/metadata.feature:53]: same comment, should this be moved above `And the build succeeded`

File: features/step_definitions/metadata.js [hunk 3]
```diff
@@ -16,13 +16,48 @@ Before(
...
+        url: `${this.instance}/${this.namespace}/pipelines/${this.pipelineId}/jobs?jobName=fourth`,
+        method: 'GET',
+        context: {
+            token: this.jwt
+        }
+    })
+        .then(resp => {
+            Assert.equal(resp.statusCode, 200);
+            Assert.equal(resp.body.length, 1);
+            Assert.equal(resp.body[0].name, 'fourth');
```
Comment (tkyi) [inline features/step_definitions/metadata.js:39]:
```suggestion
Then(/^the "(BOOZ)" job is disabled$/, { timeout: TIMEOUT }, function step(jobName) {
    return request({
        url: `${this.instance}/${this.namespace}/pipelines/${this.pipelineId}/jobs?jobName=${jobName}`,
        method: 'GET',
        context: {
            token: this.jwt
        }
    })
        .then(resp => {
            Assert.equal(resp.statusCode, 200);
            Assert.equal(resp.body.length, 1);
            Assert.equal(resp.body[0].name, jobName);
```

File: features/step_definitions/metadata.js [hunk 4]
```diff
@@ -85,3 +125,77 @@ Then(/^a record of the metadata is stored$/, { timeout: TIMEOUT }, function step
...
+                },
+                context: {
+                    token: this.jwt
+                }
+            });
+        })
+        .then(resp => {
+            Assert.equal(resp.statusCode, 200);
+        });
+});
```
Comment (tkyi) [inline features/step_definitions/metadata.js:201]: can this be removed and combined with `/^the "BOOZ" job is disabled$/`
```suggestion
```

File: features/step_definitions/metadata.js [hunk 5]
```diff
@@ -16,13 +16,48 @@ Before(
...
+        this.buildMeta = null;
         this.jwt = null;
     }
 );
 
 Given(/^a metadata starts with an empty object$/, { timeout: TIMEOUT }, () => null);
 
+Then(/^the "BOOZ" job is disabled$/, { timeout: TIMEOUT }, function step() {
+    return request({
+        url: `${this.instance}/${this.namespace}/pipelines/${this.pipelineId}/jobs?jobName=fourth`,
```
Comment (tkyi) [inline features/step_definitions/metadata.js:30]: can we avoid hardcoding `fourth`?

maybe set `this.jobName = 'fourth'` or another var?

File: features/metadata.feature (hunk 1 above)
Comment (VonnyJap) [reply features/metadata.feature:49]: This is the right order, the buildMeta is needed in order to check the metadata values. 
https://github.com/screwdriver-cd/screwdriver/blob/65f699b61cbcaf6e1b35dac44fb5a01e10ca0313/features/step_definitions/metadata.js#L96

File: features/metadata.feature (hunk 2 above)
Comment (VonnyJap) [reply features/metadata.feature:53]: This is the right order, the buildMeta is needed in order to check the metadata values. 
https://github.com/screwdriver-cd/screwdriver/blob/65f699b61cbcaf6e1b35dac44fb5a01e10ca0313/features/step_definitions/metadata.js#L96

**Review by minghay — APPROVED** [review]

File: features/metadata.feature (hunk 2 above)
Comment (tkyi) [reply features/metadata.feature:53]: Syntactically it doesn’t make sense that you add the metadata after the build is already done. Should be more like “and the metadata is added”

Comment (VonnyJap) [reply features/metadata.feature:53]: I will make sure that this is fixed through the next PR.

**Pull request merged**